src/Airship/Internal/Helpers.hs

```diff
@@ -22,12 +23,22 @@ import qualified Network.HTTP.Types        as HTTP
 import qualified Network.Wai               as Wai
 import           Network.Wai.Parse
 import           System.Random
+import           Text.Read
 
+import           Airship.Headers
 import           Airship.Internal.Decision
 import           Airship.Internal.Route
 import           Airship.Resource
 import           Airship.Types
-import           Airship.Headers
+
+-- | Reads the parameter of the provided name and attempt to convert
+-- it to the desired 'Read'-implementing type. If the parameter is not
+-- found or cannot be read successfully, the provded handler is invoed
+-- (this may often be `halt status404`).
+readParam :: Read a => Text -> Handler s m a -> Handler s m a
+readParam p def = do
+    mStr <- fmap unpack <$> HM.lookup p <$> params
+    maybe def return (mStr >>= readMaybe)
```

Comment:
I wonder if it makes sense to expose `readMaybeParam` and expose `readParam` in terms of it?